wandb/apis/public/artifacts.py

```diff
@@ -508,21 +511,216 @@ def description(self):
         """A description of the artifact collection."""
         return self._description
 
+    @description.setter
+    def description(self, description: Optional[str]) -> None:
+        self._description = description
+
     @property
     def tags(self):
         """The tags associated with the artifact collection."""
         return self._tags
 
+    @tags.setter
+    def tags(self, tags: List[str]) -> None:
+        if any(not re.match(r"^[-\w]+([ ]+[-\w]+)*$", tag) for tag in tags):
```

Comment:
question: may have missed this, so looking to understand first -- is the tag allowed to start or end with a hyphen/underscore (e.g. `"-this-is-my-tag_"`)?  If yes, understood -- just wanted to check

Reply:
If no, I think the condition/regex may need to be stricter here, since it'll consider the above example valid.  Possible, rough suggestion below:
```suggestion
        if any(not _is_valid_tag(tag) for tag in tags):
```
where:
```python3
def _is_valid_tag(tag: str) -> bool:
    split_tag = re.split(r"[ ]+|[-]+|[_]+", tag)
    return all(part.isalnum() for part in split_tag)
```

Reply:
good point, it looks like via the UI we're able to start or end tags with hyphens/underscores so I think what we have right now should be fine